packages/core/saved-objects/core-saved-objects-migration-server-internal/src/document_migrator/utils.ts

```diff
@@ -85,3 +86,21 @@ export function transformComparator(a: Transform, b: Transform) {
 
   return Semver.compare(a.version, b.version) || aPriority - bPriority;
 }
+
+/**
+ * Returns true if the given document has an higher version that the last known version, false otherwise
+ */
+export function downgradeRequired(
+  doc: SavedObjectUnsanitizedDoc,
+  latestVersions: Record<TransformType, string>
+): boolean {
+  const docTypeVersion = doc.typeMigrationVersion ?? doc.migrationVersion?.[doc.type];
+  const latestMigrationVersion = maxVersion(
+    latestVersions[TransformType.Migrate],
+    latestVersions[TransformType.Convert]
+  );
+  if (!docTypeVersion || !latestMigrationVersion) {
+    return false;
+  }
+  return Semver.gt(docTypeVersion, latestMigrationVersion);
```

Comment:
Q: is this only for V2 versions? I thought model versions didn't look like Semver, do they?

Reply:
We're using a compatibility format so that both stack versions (used by old migrations) and model versions are compatible and can share the same logic

https://github.com/elastic/kibana/blob/9cddb60324963c401ae8b88ae990a84beaab2085/packages/core/saved-objects/core-saved-objects-base-server-internal/src/model_version/version_map.ts#L13-L23